pyfr/readers/native.py

```diff
@@ -36,10 +36,22 @@ def __getitem__(self, aname):
             return self._file[aname[0]].attrs[aname[1]]
 
     def __iter__(self):
-        return iter(self._file)
+        return iter(self._keys)
 
     def __len__(self):
-        return len(self._file)
+        return len(self._keys)
+
+    @lazyprop
+    def _keys(self):
+        keys = set()
+
+        def visitor(name, item):
```

Comment:
Worth verifying that this does not cause us to read the entire data set in.  Can test easily enough with a large solution and seeing how long it takes to get the keys.

Reply:
`print([key for key in sol.keys()])` with a 10Gb solution file returned almost instantly